### PR TITLE
Feature/create dashboard page

### PR DESCRIPTION
**Pull request by LennartSchwier**
As user I want to see interesting information on a dashboard page right after logging in

- [x] current location

### CODE REVIEW

File: frontend/src/commons/Buttons.js [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
     return (
-        <ButtonGroupStyled variant="text" color="primary" aria-label="text primary button group">
-            <Button disabled={disableButtonOne} onClick={clickHandlerOne}>{labelButtonOne}</Button>
-            <Button disabled={disableButtonTwo} onClick={clickHandlerTwo}>{labelButtonTwo}</Button>
-            <Button disabled={disableButtonThree} onClick={clickHandlerThree}>{labelButtonThree}</Button>
+        <ButtonGroupStyled>
+            <ButtonStyled disabled={disableButtonOne} onClick={clickHandlerOne}>{labelButtonOne}</ButtonStyled>
+            <ButtonStyled disabled={disableButtonTwo} onClick={clickHandlerTwo}>{labelButtonTwo}</ButtonStyled>
+            <ButtonStyled disabled={disableButtonThree} onClick={clickHandlerThree}>{labelButtonThree}</ButtonStyled>
+            <ButtonStyled disabled={disableButtonFour} onClick={clickHandlerFour}>{lableButtonFour}</ButtonStyled>
```
Comment (jamarob) [inline frontend/src/commons/Buttons.js:12]: Can you find a better name for this component? Just from reading the code I have no clue what they are used for. Do you really need a totally generic group of 4 buttons? Imho having one button component and rendering it where and when it is needed is preferable.

File: frontend/src/components/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    const history = useHistory();
+
+    const [positionAvailable, latitude, longitude, errorMessage] = usePositioning();
+
+    return (
+        <PageLayout>
+            <Header headerText={"Ahoi 'username'"}/>
+            <PositionBlock>
+                <div>Current location: {positionAvailable ? <b>Available</b> : <b>Not Available</b>}</div>
```
Comment (jamarob) [inline frontend/src/components/Dashboard.js:18]: Better use a `section`, `div`s are just for styling. Use of `b` is dicouraged, style with css.

File: frontend/src/components/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                {positionAvailable ?
+                    <div>
+                        <div>Latitude: <b>{latitude}</b></div>
+                        <div>Longitude: <b>{longitude}</b></div>
+                    </div>
+                    :
+                    <div id={"error"}>{errorMessage}</div>
```
Comment (jamarob) [inline frontend/src/components/Dashboard.js:25]: There seems to be a coupling between `errorMessage` and `positionAvailable`. Maybe you can use that to make your code more descriptive? Can there be different error messages if `positionAvailable` is falsy?

File: frontend/src/components/Dashboard.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+`
+
+const PositionBlock = styled.div`
+margin: 0 var(--size-l);
+
+  div {
+  margin: var(--size-s);
+  }
+  
+  #error {
```
Comment (jamarob) [inline frontend/src/components/Dashboard.js:58]: 😱 never ever style with ids!! use a class instead!

File: frontend/src/components/Dashboard.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                disableButtonOne={false}
+                labelButtonOne={"Log Out"}
+                disableButtonThree={false}
+                labelButtonThree={"Input"}
+                clickHandlerThree={clickHandlerInput}
+            />
+        </PageLayout>
+    );
+
+    function clickHandlerInput() {
```
Comment (jamarob) [inline frontend/src/components/Dashboard.js:39]: Try a verb oriented naimng style for functions, better say what it does, not what it is.

File: frontend/src/components/HeadSail.js [hunk 6]
```diff
@@ -70,7 +70,7 @@ justify-content: space-evenly;
 align-items: center;
 font-size: 0.9em;
 
-  div > div {
+  div > span {
```
Comment (jamarob) [inline frontend/src/components/HeadSail.js:73]: Maybe remove the duplication and introduce a class or styled component? Relying on component nesting for styling is not very robust.

File: frontend/src/components/Input.js [hunk 7]
```diff
@@ -62,6 +64,10 @@ export default function Input({course, setCourse, windSpeed, setWindSpeed, waveH
         setCourse(event.target.value)
     }
 
+    function clickHandlerDashboard() {
```
Comment (jamarob) [inline frontend/src/components/Input.js:67]: Verb oriented names please.

File: frontend/src/components/MainSail.js [hunk 8]
```diff
@@ -78,7 +78,7 @@ justify-content: space-evenly;
 align-items: center;
 font-size: 0.9em;
 
-  div > div {
+  div > span {
```
Comment (jamarob) [inline frontend/src/components/MainSail.js:81]: I have seen this somewhere else :)

File: frontend/src/hooks/usePositioning.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    const geoError = (error) => {
+        setErrorMessage("Error: Code " + error.code + " - " + error.message);
+    }
+
+    useEffect(() => {
+        const geoOptions = {
+            enableHighAccuracy: true,
+            maximumAge: 30000,
+            //timeout: 35000
```
Comment (jamarob) [inline frontend/src/hooks/usePositioning.js:25]: please delete comments

File: frontend/src/hooks/usePositioning.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        console.log("checking availability...")
```
Comment (jamarob) [inline frontend/src/hooks/usePositioning.js:27]: please delete console.log

File: frontend/src/hooks/usePositioning.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        if (!navigator.geolocation) {
+            setGeoAvailable(false)
+        }
+        else {
+            console.log("requesting position...");
```
Comment (jamarob) [inline frontend/src/hooks/usePositioning.js:32]: please delete

File: frontend/src/hooks/usePositioning.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+export default function usePositioning() {
+
+
+    const [geoAvailable, setGeoAvailable] = useState(false);
+    const [latitude, setLatitude] = useState(null);
+    const [longitude, setLongitude] = useState(null);
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const geoSuccess = (position) => {
+        setGeoAvailable(true);
```
Comment (jamarob) [inline frontend/src/hooks/usePositioning.js:12]: I think you can get rid of the `geoAvailable` state as it can be derived as `latitide && longitude`, no?

**Review by jamarob — COMMENTED** [review]
I left some comments regarding naming and simplification. Good job!

File: backend/src/main/java/de/neuefische/saildog/utils/MainSailTrimUtils.java [hunk 13]
```diff
@@ -184,16 +184,24 @@ public MainSail calculateMainSailTrim(BoatCourse boatCourse, double wind, double
...
+                return mainSailBuilder(
+                        SheetState.LOOSE, TravellerState.LEE,
+                        BoomVangState.MAX_TIGHT, LuffFootState.SMOOTH,
+                        LuffFootState.SLIGHTLY_CLOSED);
             }
         }
         return new MainSail();
     }
+
+    private MainSail mainSailBuilder(SheetState sheetState, TravellerState travellerState,
```
[the remark this answers is not in the document]
Comment (LennartSchwier) [reply backend/src/main/java/de/neuefische/saildog/utils/MainSailTrimUtils.java:196]: Rikardo made then suggestion to write a helper function and I wanted to try it out once before refactoring the whole class. But I rather tend to stick to the standard builder

File: frontend/src/commons/Buttons.js (hunk 1 above)
Comment (LennartSchwier) [reply frontend/src/commons/Buttons.js:12]: Good point! :)

File: frontend/src/components/Dashboard.js (hunk 2 above)
Comment (LennartSchwier) [reply frontend/src/components/Dashboard.js:18]: But what is best practice for only having a part of a text as bold text?

Comment (jamarob) [reply frontend/src/components/Dashboard.js:18]: could be something as simple as `const Bold = styled.span`, using `b` is discouraged as it conveys no semantic meaning and we want to restrict html to semantics not styling

File: frontend/src/components/Dashboard.js (hunk 4 above)
Comment (LennartSchwier) [reply frontend/src/components/Dashboard.js:58]: ok 🙈

**Pull request merged**